src/modules/mc_pos_control/PositionControl.cpp

```diff
@@ -285,58 +263,23 @@ void PositionControl::_velocityController(const float &dt)
 
 void PositionControl::updateConstraints(const Controller::Constraints &constraints)
 {
-	// update all parameters since they might have changed
-	_updateParams();
-
 	_constraints = constraints;
 
-	// Check if adustable contraints are below global constraints. If they are not stricter than global
+	// Check if adjustable constraints are below global constraints. If they are not stricter than global
 	// constraints, then just use global constraints for the limits.
-	if (!PX4_ISFINITE(constraints.tilt_max) || !(constraints.tilt_max < MPC_TILTMAX_AIR)) {
-		_constraints.tilt_max = MPC_TILTMAX_AIR;
-	}
 
-	if (!PX4_ISFINITE(constraints.vel_max_z_up) || !(constraints.vel_max_z_up < MPC_Z_VEL_MAX_UP)) {
-		_constraints.vel_max_z_up = MPC_Z_VEL_MAX_UP;
+	if (!PX4_ISFINITE(constraints.tilt_max) || !(constraints.tilt_max < MPC_TILTMAX_AIR.get())) {
+		_constraints.tilt_max = MPC_TILTMAX_AIR.get();
 	}
-}
 
-void PositionControl::_updateParams()
-{
-	bool updated;
-	parameter_update_s param_update;
-	orb_check(_parameter_sub, &updated);
-
-	if (updated) {
-		orb_copy(ORB_ID(parameter_update), _parameter_sub, &param_update);
-		_setParams();
+	if (!PX4_ISFINITE(constraints.vel_max_z_up) || !(constraints.vel_max_z_up < MPC_Z_VEL_MAX_UP.get())) {
+		_constraints.vel_max_z_up = MPC_Z_VEL_MAX_UP.get();
 	}
 }
 
-void PositionControl::_setParams()
+void PositionControl::overwriteParams()
 {
-	param_get(MPC_XY_P_h, &_Pp(0));
-	param_get(MPC_XY_P_h, &_Pp(1));
-	param_get(MPC_Z_P_h, &_Pp(2));
-	param_get(MPC_XY_VEL_P_h, &_Pv(0));
-	param_get(MPC_XY_VEL_P_h, &_Pv(1));
-	param_get(MPC_Z_VEL_P_h, &_Pv(2));
-	param_get(MPC_XY_VEL_I_h, &_Iv(0));
-	param_get(MPC_XY_VEL_I_h, &_Iv(1));
-	param_get(MPC_Z_VEL_I_h, &_Iv(2));
-	param_get(MPC_XY_VEL_D_h, &_Dv(0));
-	param_get(MPC_XY_VEL_D_h, &_Dv(1));
-	param_get(MPC_Z_VEL_D_h, &_Dv(2));
-	param_get(MPC_XY_VEL_MAX_h, &MPC_XY_VEL_MAX);
-	param_get(MPC_Z_VEL_MAX_UP_h, &MPC_Z_VEL_MAX_UP);
-	param_get(MPC_Z_VEL_MAX_DN_h, &MPC_Z_VEL_MAX_DN);
-	param_get(MPC_THR_HOVER_h, &MPC_THR_HOVER);
-	param_get(MPC_THR_MAX_h, &MPC_THR_MAX);
-	param_get(MPC_THR_MIN_h, &MPC_THR_MIN);
-	param_get(MPC_MANTHR_MIN_h, &MPC_MANTHR_MIN);
-	param_get(MPC_TILTMAX_AIR_h, &MPC_TILTMAX_AIR);
-	MPC_TILTMAX_AIR = math::radians(MPC_TILTMAX_AIR);
-	param_get(MPC_MAN_TILT_MAX_h, &MPC_MAN_TILT_MAX);
-	MPC_MAN_TILT_MAX = math::radians(MPC_MAN_TILT_MAX);
-
+	// Tilt needs to be in radians
+	MPC_TILTMAX_AIR.set(math::radians(MPC_TILTMAX_AIR.get()));
```

Comment:
This is prone to errors: if for some reason `overwriteParams` is not called, or several times, or you use `MPC_TILTMAX_AIR.commit()`, you will have completely wrong values. I suggest using a separate class member instead (I used the `_deg` suffix to make clear which one is in degrees). Not super nice, but much less error prone.